src/span-link.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Builds a payload for a Step Function LambdaInvoke task, so the Step Function traces
+ * can be merged with downstream Lambda traces.
+ *
+ * This function modifies the provided payload to include context fields necessary
+ * for trace merging purposes. If the payload already contains any of these fields,
+ * an error is thrown to avoid conflicts.
+ *
+ * @param payload - The user's payload object. Defaults to an empty object.
+ * @returns The modified payload object with necessary context added.
+ * @throws {ConflictError} If the payload already contains `Execution`, `State`, or `StateMachine` fields.
+ */
+export function buildStepFunctionLambdaTaskPayloadToMergeTraces(payload: { [key: string]: any } = {}): {
+  [key: string]: any;
```

Comment:
If customer's `payload` is something custom like a String, what's the failure mode?

Does the TS type checker throw an Error?